systemtest/src/test/java/io/strimzi/systemtest/cruisecontrol/CruiseControlST.java

```diff
@@ -639,6 +642,141 @@ void testKafkaRebalanceAutoApprovalMechanism() {
+                knp.getSpec().setReplicas(scaleTo));
+        } else {
+            KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(scaleTo));
+        }
+
+        // 3 brokers + 3 controllers/zk
+        final int kafkaClusterPodIndex = initialReplicas + initialReplicas;
+
+        KafkaUtils.waitUntilKafkaHasExpectedModeAndBrokersAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(),
+            KafkaRebalanceMode.ADD_BROKERS,
+            // brokers with [6, 7]
+            Arrays.asList(kafkaClusterPodIndex, kafkaClusterPodIndex + 1));
+
+        // check that Kafka has `RebalanceOnScaleUp` in auto-rebalance status
+        KafkaUtils.waitUntilKafkaHasStateAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaAutoRebalanceState.RebalanceOnScaleUp);
+
+        // and then afterward we again move to Idle state after all is done
+        KafkaUtils.waitUntilKafkaHasStateAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(), KafkaAutoRebalanceState.Idle);
+
+        // CC Rolled
+        ccPod = DeploymentUtils.waitTillDepHasRolled(testStorage.getNamespaceName(), CruiseControlResources.componentName(testStorage.getClusterName()), 1, ccPod);
+
+        // we double-check that Rolling has been done.
+        RollingUpdateUtils.waitForComponentScaleUpOrDown(testStorage.getNamespaceName(), testStorage.getBrokerSelector(), scaleTo);
+
+        LOGGER.info("Checking that Topic: {} has replicas on one of the new brokers (or both)", testStorage.getTopicName());
+        List<String> topicReplicas = KafkaTopicUtils.getKafkaTopicReplicasForEachPartition(testStorage.getNamespaceName(), testStorage.getTopicName(), scraperPodName, KafkaResources.plainBootstrapAddress(testStorage.getClusterName()));
+        assertTrue(topicReplicas.stream().anyMatch(line -> line.contains("6") || line.contains("7")));
+
+        LOGGER.info("Scaling Kafka down to {}", initialReplicas);
+
+        if (Environment.isKafkaNodePoolsEnabled()) {
+            KafkaNodePoolResource.replaceKafkaNodePoolResourceInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getBrokerPoolName(), knp ->
+                knp.getSpec().setReplicas(initialReplicas));
+        } else {
+            KafkaResource.replaceKafkaResourceInSpecificNamespace(testStorage.getNamespaceName(), testStorage.getClusterName(), kafka -> kafka.getSpec().getKafka().setReplicas(initialReplicas));
+        }
+
+        KafkaUtils.waitUntilKafkaHasExpectedModeAndBrokersAutoRebalance(testStorage.getNamespaceName(), testStorage.getClusterName(),
+            KafkaRebalanceMode.REMOVE_BROKERS,
+            // brokers with [6, 7]
+            Arrays.asList(kafkaClusterPodIndex, kafkaClusterPodIndex + 1));
+
```

Comment:
I would also add the check that the corresponding KafkaRebalance custom resource for CC was created.